Apps/Src/Temperature.c

```diff
@@ -420,22 +413,26 @@ int32_t Temperature_GetMaxTemperature(void) {
  * @brief Gives fan speed based on Average temperature of pack and past error values
  * @param InputTemp - current temperature
  * @param DesiredTemp - desired temperature
- * @return FanSpeed: 0-4000 PWM
+ * @return FanSpeed: 0-8
  */
 int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
     Error = DesiredTemp - InputTemp;
-    ErrorSum = ErrorSum + Error;
+	
+    //Only read error sum in range
+    ErrorSum += abs(Error) < TEMPERATURE_PID_I_ZONE ? Error : 0;
 
-    if (PreviousError == 0) {PreviousError = Error;} //init previous val first time
+	//Cap error sum at 500 read degrees
+    ErrorSum = abs(ErrorSum) > TEMPERATURE_PID_I_MAX_ACCUM ? ErrorSum > 0 ? TEMPERATURE_PID_I_MAX_ACCUM : -TEMPERATURE_PID_I_MAX_ACCUM : ErrorSum;
 
-    Rate = Error - PreviousError;
-    PreviousError = Error;     //updates previous err value
+	//5 is the estimated "hold output"
```

Comment:
you should still have the 5 as a define at the top of the file

Reply:
fixed